### PR TITLE
Improved meta-data for models

### PR DESCRIPTION
**Pull request by datumbox**
Enhance the meta-data information stored for each model by including the following info:
- Type of task
- Network architecture
- Year of publication
- Number of parameters

cc @datumbox @bjuncek

### CODE REVIEW

File: torchvision/prototype/models/detection/mask_rcnn.py [hunk 1]
```diff
@@ -28,6 +28,9 @@ class MaskRCNN_ResNet50_FPN_Weights(WeightsEnum):
         url="https://download.pytorch.org/models/maskrcnn_resnet50_fpn_coco-bf2d0c1e.pth",
         transforms=CocoEval,
         meta={
+            "task": "image_object_detection",
+            "architecture": "FasterRCNN",
+            "publication_year": 2015,
```
Comment (fmassa) [inline torchvision/prototype/models/detection/mask_rcnn.py:33]: Mask R-CNN was published in a different paper than Faster R-CNN (2017)

File: torchvision/prototype/models/detection/mask_rcnn.py [hunk 2]
```diff
@@ -28,6 +28,9 @@ class MaskRCNN_ResNet50_FPN_Weights(WeightsEnum):
         url="https://download.pytorch.org/models/maskrcnn_resnet50_fpn_coco-bf2d0c1e.pth",
         transforms=CocoEval,
         meta={
+            "task": "image_object_detection",
+            "architecture": "FasterRCNN",
```
Comment (fmassa) [inline torchvision/prototype/models/detection/mask_rcnn.py:32]: I would say the `architecture` for this one should probably be `MaskRCNN`, as it's an extension of Faster R-CNN which has its own paper https://paperswithcode.com/paper/mask-r-cnn

File: torchvision/prototype/models/detection/mask_rcnn.py [hunk 3]
```diff
@@ -28,6 +28,9 @@ class MaskRCNN_ResNet50_FPN_Weights(WeightsEnum):
         url="https://download.pytorch.org/models/maskrcnn_resnet50_fpn_coco-bf2d0c1e.pth",
         transforms=CocoEval,
         meta={
+            "task": "image_object_detection",
```
Comment (fmassa) [inline torchvision/prototype/models/detection/mask_rcnn.py:31]: I'm unclear on what to put here.
Indeed, this model could also be categorized as `image_instance_segmentation`. Maybe having `task` be a list would be of help here?

Ensuring that this is adapted with paperswithcode categorization would be good as well https://paperswithcode.com/paper/mask-r-cnn

File: torchvision/prototype/models/detection/keypoint_rcnn.py [hunk 4]
```diff
@@ -24,6 +24,9 @@
 
 
 _COMMON_META = {
+    "task": "image_object_detection",
+    "architecture": "FasterRCNN",
+    "publication_year": 2015,
```
Comment (fmassa) [inline torchvision/prototype/models/detection/keypoint_rcnn.py:29]: Same comments as for MaskRCNN -- KeypointRCNN was introduced in the Mask R-CNN paper. In this case you could potentially use the same `architecture` though, although I could also see it being its own architecture

File: torchvision/prototype/models/detection/mask_rcnn.py (hunk 3 above)
Comment (datumbox) [reply torchvision/prototype/models/detection/mask_rcnn.py:31]: Yeah same here. I'll leave as is for now and review in more detail on the future.

**Review by fmassa — APPROVED** [review]
LGTM

File: torchvision/prototype/models/detection/keypoint_rcnn.py [hunk 5]
```diff
@@ -24,6 +24,9 @@
 
 
 _COMMON_META = {
+    "task": "image_object_detection",
```
Comment (oke-aditya) [inline torchvision/prototype/models/detection/keypoint_rcnn.py:27]: image keypoint detection. I guess?

**Comment by oke-aditya** [comment]
Could we add size of the model in MBs #4996 ? Anything about FLOPS of computation / Runtime statistics too would be cool.

(Although it's hard to say as runtime highly various from system to system and configs)

**Comment by datumbox** [comment]
@oke-aditya Sounds like good suggestions. Would you like to send a PR that adds model sizes in? Ideally with a unit-test to compare the declared meta-data size with the actual size (you are guaranteed that they are available locally but you might need to be creative on how to do this; I got some similar checks here in this PR for model size in params).

FLOPS would be really cool, but it's not always straightforward to automatically estimate this for all models (for example detection models). I'm open for any ideas. BTW we've got a feedback issue at #5088 were we keep proposals about the new API centralized. Feel free to post more there if you have additional ideas. :)

**Comment by oke-aditya** [comment]
I would need some time for this as it's been hectic week at office :( (It's literally odd to code in Java in morning and python at night)

**Pull request merged**